builtin/mainmenu/tab_online.lua

```diff
@@ -285,14 +361,9 @@ local function search_server_list(input)
 	-- Search the serverlist
 	local search_result = {}
 	for i, server in ipairs(serverlistmgr.servers) do
-		local name_matches, description_matches = true, true
-		for _, keyword in ipairs(keywords) do
-			name_matches = name_matches and not not
-					(server.name or ""):lower():find(keyword, 1, true)
-			description_matches = description_matches and not not
-					(server.description or ""):lower():find(keyword, 1, true)
-		end
-		if name_matches or description_matches then
+		local filter_matches, name_matches, description_matches = matches_query(server, query)
+
+		if filter_matches and (name_matches or description_matches) then
```

Comment:
```suggestion
		if matches_query(server, query) then
```

(and the two lines above this can go)